### DISCUSSION TITLE
Music problem - Won't play midi or midi expansion options

### DISCUSSION THREAD

**Discussion by amirshac** [ihhub/fheroes2#8601] [Q&A]
Hi forum,
Music problem - Won't play midi or midi expansion options
Only "external" music option works

OS: Linux Mint 21.3
Fheroes version: 1.0.13 flatpack

I've used the 'manual' method of copying GOG directories to ~/.var/app/io.github.ihhub.Fheroes2/data/fheroes2

I only have .ogg files under music directory

Couldn't find any .midi files

Any way to troubleshoot this?

Thank you

Screenshot:
![fheroesmidi](https://github.com/ihhub/fheroes2/assets/76565122/a90bd549-0a6e-496b-9577-61c9d2ecb9fa)

**Comment by zenseii** [comment]
Hi, @amirshac. This might be because you're missing a midi soundfont. Perhaps the flatpak package doesn't have this setup properly. 

Would you happen to know something about this, @Laserlicht ?

**Reply by amirshac** [reply]
Just tried installing the game manually (.zip version from the downloads)
midi music does work on that version

So it does seem to be an issue with the flatpak version of the game.

**Reply by Laserlicht** [reply]
I don't know how MIDI playback works under Linux, nor under fheroes2. I have never dealt with it before. :(

Maybe @undef21 can help? He contributed a patch to support MIDI in our Flatpak: https://github.com/flathub/io.github.ihhub.Fheroes2/pull/21

**Comment by undef21** [comment]
@amirshac 
Could you run `flatpak run io.github.ihhub.Fheroes2` in a terminal, set MIDI music in options and place output here?

**Reply by amirshac** [reply]
oh! ofcourse!

`[ERROR]	createMusic:  Failed to create a music track from memory. The error: Unrecognized audio format`

That explains it!

**Comment by undef21** [comment]
Hello.
@Laserlicht , it looks like `freedesktop-sdk`  [dropped](https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/blob/master/elements/components/sdl2-mixer.bst?ref_type=heads#L18) MIDI support for `sdl_mixer`. So you should use your own `libsdl2_mixer` + soundfonts for your flatpak or `freedesktop-sdk` developers should enable MIDI in their library.

**Reply by Laserlicht** [reply]
@nanonyme What was the reason for removing? Is it possible to enable it again?
